<?xml version="1.0"?>
<robot  xmlns:xacro="http://www.ros.org/wiki/xacro" name="firefighter" >
	
<xacro:property name="width" value=".2" />


  <link name="joint1">
    <visual>
      <geometry>
	     <!--- 0.0 0 -0.04 -->
       <mesh filename="package://mycobot_ros/urdf/joint1.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
	<material name = "white">
		<color rgba = "0.1 0.1 0.9 1"/>
	</material>
    </visual>
  </link>

  <link name="joint2">
    <visual>
      <geometry>
       
       <mesh filename="package://mycobot_ros/urdf/joint2.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
  <material name = "white">
    <color rgba = "0.4 0.4 0.4 1"/>
  </material>
    </visual>
  </link>


  <link name="joint3">
    <visual>
      <geometry>
       
       <mesh filename="package://mycobot_ros/urdf/joint3.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
  <material name = "white">
    <color rgba = "0.4 0.4 0.4 1"/>
  </material>
    </visual>
  </link>


  <link name="joint4">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://mycobot_ros/urdf/joint4.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
  <material name = "white">
    <color rgba = "0.4 0.4 0.4 1"/>
  </material>
    </visual>
  </link>




<link name="joint5">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://mycobot_ros/urdf/joint5.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
  <material name = "white">
    <color rgba = "0.4 0.4 0.4 1"/>
  </material>
    </visual>
  </link>


  <link name="joint6">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://mycobot_ros/urdf/joint6.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
  <material name = "white">
    <color rgba = "0.4 0.4 0.4 1"/>
  </material>
    </visual>
  </link>


  <link name="joint6_flange">
    <visual>
      <geometry>

        <cylinder length="0.005" radius="0.02"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 1.57075 0 0"/>
  <material name = "white">
    <color rgba = "0.9 0.9 0.9 1"/>
  </material>
    </visual>
  </link>



    <joint name="joint2_to_joint1" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="joint1"/>
    <child link="joint2"/>
    <origin xyz= "0 0 0.0706" rpy = "0 0 0"/>  
  </joint>

  <joint name="joint3_to_joint2" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="joint2"/>
    <child link="joint3"/>
    <origin xyz= "0.03256 0  0.05976" rpy = "0 0 0"/>  
  </joint>

  <joint name="joint4_to_joint3" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="joint3"/>
    <child link="joint4"/>
    <origin xyz= " 0.00272 0 0.1104   " rpy = "0 0 0"/>  
  </joint>

  <joint name="joint5_to_joint4" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="joint4"/>
    <child link="joint5"/>
    <origin xyz= "-0.00164 0 0.096" rpy = "0 0 0"/>  
  </joint>

  <joint name="joint6_to_joint5" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="joint5"/>
    <child link="joint6"/>
    <origin xyz= "0.0288 0 0.02976" rpy = "0 0 0"/>  
  </joint>

  <joint name="joint6output_to_joint6" type="revolute">
    <axis xyz="0 1  0"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="joint6"/>
    <child link="joint6_flange"/>
    <origin xyz= "0 0.0456 0.03792 " rpy = "0 0 0"/>  
  </joint>



</robot>

